### PR TITLE
Update Jekyll, add sitemap, other site stuff

### PR DESCRIPTION
**Pull request by jpstroop**
* Add sitemap plugin and sitemap link to robots.txt
 * Switch from pygments to rouge (Jekyll's new default, has `json-doc` syntax that allows for comments)
 * Use jekyll-redirect-from gem instead of custom plugin

This PR won't build on the site without some intervention. To test drive locally, do:

```
bundle install
./dev.sh
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,5 +18,5 @@ Markdown Source of specifications documents
 ## Some Things to Note
 
  * Much of the site data is in the YAML files in `_data/` (e.g. member institutions, server impls, demos, etc.) make edits there.
- * The latest versions of the APIs are set in `_config.yml`. Change there will get pushed to `.htaccess`, `technical-details.html`, and any other links.
+ * The latest versions of the APIs are set in `_config.yml`. Change there will get pushed to `.htaccess`, `technical-details/`, and any other links.
```
Comment (jpstroop) [inline README.md]: Not how to static HTML pages are now directories, e.g. `/technical-details.html` :arrow_right: `/technical-details/index.html`

**Comment by zimeon** [comment]
@jpstroop: do you know what URIs the change of /technical-details.html :arrow_right: /technical-details/index.html style might break?

**Comment by jpstroop** [comment]
@zimeon I updated all of the links within the site, and the tests check all links, so I think we're good. I suppose adding a redirect from /technical-details.html :arrow_right: /technical-details/index.html wouldn't hurt. Shall I?

**Comment by jpstroop** [comment]
I don't think this hurts any LD URIs, if that's what you were asking.

**Comment by zimeon** [comment]
I was asking generally about any URI changes that might cause breakage. Good to hear that LD URIs are OK.


I like the Jekyll change to localize redirect information with the target file. I see the removal of [source/_redirects.json](https://github.com/IIIF/iiif.io/blob/c34b8ad51aab244fa3f8cc58929adf6075367434/source/_redirects.json) and addition of extra redirect headers like:

```
redirect_from:
  - /api/image/1/index.html
```

I would seem sensible to add redirects for the 4 static files affected by the URI changes:

```
/about.html -> /about/
/technical-details.html -> /technical-details/
/apps-demos.html -> /apps-demos/
/community.html -> /community/
```

I'm not sure it is good practice to tidy old specs, there seems no real benefit to fixing indents in: [source/model/shared-canvas/1.0/index.html](https://github.com/IIIF/iiif.io/commit/09eace54d4644f5c1f3895f53ead60a39ff9e370#diff-f6d73ec156e4c19558de9d1c2a5f6b68) but there is the possibility of causing confusion and (though I don't see any) introducing errors.

**Comment by zimeon** [comment]
Oh, and we should make sure the entire Stanford fire dept is online before any attempt to merge this one!!!!

**Comment by jpstroop** [comment]
The latest Jekyll wouldn't build the shared canvas spec without some character cleanup.

**Comment by jpstroop** [comment]
New static links are there.

**Comment by azaroth42** [comment]
Merging because better than being dead.

**Pull request merged**